File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        m=prices[0]
        d=0
        c=0
        for i in range(1,len(prices)):
            m=min(m,prices[i])
            if prices[i]<prices[i-1]:
                c+=1 
            else:
                diff=prices[i]-m
                d=max(d,diff)
                
        if c==len(prices):
            return 0
        else:
            return d

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0;
        int start=0;
        for(int end=1;end<prices.size();end++){
            cout<<start<<" "<<end<<endl;
            if(prices[start]>prices[end])start=end;
            profit=max(profit,prices[end]-prices[start]);
        }
        return profit;
    }
};